python/pyarrow/fs.py

```diff
@@ -227,16 +227,28 @@ def copy_files(source, destination,
 
     Examples
     --------
-    Copy an S3 bucket's files to a local directory:
+    Inspect an S3 bucket's files:
 
-    >>> copy_files("s3://your-bucket-name",
-    ...            "local-directory") # doctest: +SKIP
+    >>> s3, path = fs.FileSystem.from_uri(
+    ...            "s3://registry.opendata.aws/roda/ndjson/")
+    >>> selector = fs.FileSelector(path)
+    >>> s3.get_file_info(selector)
+    [<FileInfo for 'registry.opendata.aws/roda/ndjson/index.ndjson':...]
 
-    Using a FileSystem object:
+    Copy one file from S3 bucket to a local directory:
 
-    >>> copy_files("your-bucket-name", "local-directory",
-    ...            source_filesystem=S3FileSystem(...)) # doctest: +SKIP
+    >>> fs.copy_files("s3://registry.opendata.aws/roda/ndjson/index.ndjson",
+    ...               "file:///{}/index_copy.ndjson".format(local_path))
 
+    >>> selector2 = fs.FileSelector(str(local_path))
+    >>> local.get_file_info(selector2)
```

Comment:
Corrected the proposed change.
Didn't explicitly created a local filesystem as now only `local_path` is included and I hope that from the name it can be understood what the path links to. Tried to focus more on `copy_files` method as I agree it makes more sense.